perceval/backends/mozilla/mozillaclub.py

```diff
@@ -0,0 +1,397 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+
+
+from ...backend import Backend, BackendCommand, metadata
+from ...cache import Cache
+from ...errors import BackendError, CacheError, ParseError
+
+from ...utils import (DEFAULT_DATETIME,
+                     datetime_to_utc,
+                     str_to_datetime,
+                     urljoin)
+
+
+logger = logging.getLogger(__name__)
+
+MOZILLA_CLUB_URL = "https://spreadsheets.google.com/feeds/cells/1QHl2bjBhMslyFzR5XXPzMLdzzx7oeSKTbgR5PM8qp64/ohaibtm/public/values?alt=json"
+
+EVENT_TEMPLATE = {
+    "1": "Status",
+    "2": "Date of Event",
+    "3": "Club Name",
+    "4": "Country",
+    "5": "Your Name",
+    "6": "Your Twitter Handle (Optional)",
+    "7": "Event Description",
+    "8": "Attendance",
+    "9": "Club Link",
+    "10": "City",
+    "11": "Event Creations",
+    "12": "Web Literacy Skills",
+    "13": "Links to Curriculum (Optional)",
+    "14": "Links to Blogpost (Optional)",
+    "15": "Links to Video (Optional)",
+    "16": "Links to Photos (Optional)",
+    "17": "Feedback from Attendees",
+    "18": "Your Feedback",
+    "19": "Timestamp",
+    "20": "Event Cover Photo"
+}
+
+
+class MozillaClub(Backend):
+    """MozillaClub backend for Perceval.
+
+    This class retrieves the data from MozillaClub.
+
+    :param url: Mozilla Club Events url
+    :param cache: cache object to store raw data
+    :param tag: label used to mark the data
+    """
+    version = '0.1.0'
+
+    def __init__(self, url=MOZILLA_CLUB_URL, cache=None, tag=None):
+        origin = url
+        self.url = url
+        super().__init__(origin, tag=tag, cache=cache)
+        self.client = MozillaClubClient(url)
+        self.__users = {}  # internal users cache
+
+    @metadata
+    def fetch(self):
+        """Fetch events from the MozillaClub url.
+
+        The method retrieves, from a MozillaClub url, the
+        events. The data is a Google spreadsheet retrieved using
+        the feed API REST.
+
+        :returns: a generator of events
+        """
+
+        logger.info("Looking for events at url '%s'", self.url)
+
+        nevents = 0  # number of events processed
+        nevents_wrong = 0  # number of events with wrong data
+
+        self._purge_cache_queue()
+
+        raw_cells = self.client.get_cells()
+        self._push_cache_queue(raw_cells)
+        sheet_json = json.loads(raw_cells)
+        self._flush_cache_queue()
```

Comment:
This statement should be executed once the parsed has returned all the events. Otherwise it would lead to a cache with invalid data when the parser fails.